ssg/entities/profile_base.py

```diff
@@ -259,6 +259,14 @@ def unselect_empty_groups(self, root_group):
             raise ValueError(msg)
         self.unselected_groups.extend(sorted(empty_groups))
 
+    def remove_components_not_included(self, components_to_not_include):
+        rules_to_not_include = components_to_not_include.get("rules", set())
+        groups_to_not_include = components_to_not_include.get("groups", set())
+        self.selected = list(set(self.selected) - rules_to_not_include)
```

Comment:
I have a small ask. Before this change the selects in the Profile element in the built XML were sorted nicely. Now they are random, probably due to usage of sets here. I admit we're not testing it and it isn't the case for all elements.  But, having the items sorted and ordered deterministically helps us when manual reviewing 2 different data streams. So it would be nice if this is preserved. You can either re-sort it here in this method or you can find a better place, I'm guessing sorting the items in the `_add_selects` method could also work.